src/main/java/com/softserve/academy/spaced/repetition/domain/Course.java

```diff
@@ -133,19 +136,33 @@ public void setCourseRatings(List<CourseRating> courseRatings) {
         this.courseRatings = courseRatings;
     }
 
+    public List<CourseComment> getCourseComments() {
+        return courseComments;
+    }
+
+    public void setCourseComments(List<CourseComment> courseComments) {
+        this.courseComments = courseComments;
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
-        if (!(o instanceof Course)) return false;
+        if (o == null || getClass() != o.getClass()) return false;
 
         Course course = (Course) o;
 
         if (Double.compare(course.rating, rating) != 0) return false;
+        if (published != course.published) return false;
         if (id != null ? !id.equals(course.id) : course.id != null) return false;
         if (name != null ? !name.equals(course.name) : course.name != null) return false;
         if (description != null ? !description.equals(course.description) : course.description != null) return false;
         if (image != null ? !image.equals(course.image) : course.image != null) return false;
-        return category != null ? category.equals(course.category) : course.category == null;
+        if (owner != null ? !owner.equals(course.owner) : course.owner != null) return false;
```

Comment:
Please do not add this next time. The id of an entitity is its unique identifier (a relational database guarantees this), it's enough to compare ids. The same for the hash code.